Sources/Shared/Data Manager/Model/Documents/File.m

```diff
@@ -74,8 +78,54 @@ - (void)dealloc {
     [_creator release];
     [_dateCreated release];
     [_dateModified release];
+    _naturalName = nil;
     [super dealloc];
 }
 
+-(NSString *) getNaturalName{
+    if (_naturalName!=nil) {
+        return _naturalName;
+    }
+    return _name;
+    
+}
+
+/*
+ In case of Group section, the name of the folder are genered automatically by the system & this names are not really nice (nature). The goal of this method is to convert this kind of name
+ */
+-(NSString* ) convertToNaturalName {
+    
+    NSString * s = _name;
```

Comment:
It is better to use NSMutableString to perform operations on strings:
```
NSMutableString * s = [_name mutableCopy];
[s replaceOccurrencesOfString ...];
```
I made a quick test, the current implementations takes ~10s to perform 1,000,000 calls to this method.
See the code to run this test https://gist.github.com/paristote/2966a03af7be9ba20d22.